client/client.go

```diff
@@ -267,7 +267,7 @@ func (cli *Client) NegotiateAPIVersionPing(p types.Ping) {
 	}
 
 	// if server version is lower than the maximum version supported by the Client, downgrade
-	if versions.LessThan(p.APIVersion, api.DefaultVersion) {
+	if versions.LessThan(p.APIVersion, cli.version) {
```

Comment:
I would assume that if `cli.version!=api.DefaultVersion` then user has explicitly wanted to override. Both of these variables are defined in client side, not like `p.APIVersion` that client can't predict. If we want to disable anything greater than `DefaultVersion` then the `&&` condition is not correct here and `cli.version` should be first capped to default instead.